File: app/src/main/java/io/recebimento/network/ApiService.kt
package io.recebimento.network

import okhttp3.ResponseBody
import retrofit2.Response
import retrofit2.http.*

interface ApiService {

    // ========== RECEBIMENTOS ==========
    @GET("web/recebimento/{storeId}")
    suspend fun getRecebimentos(
        @Path("storeId") storeId: String,
        @Query("status") status: String,
        @Query("search") search: String? = null,
        @Query("sort") sort: String = "asc",
        @Query("page") page: Int = 1
    ): Response<RecebimentoResponse>

    // ========== DETALHES DA VIAGEM ==========
    @GET("web/recebimento/{storeId}/{viagemId}")
    suspend fun getDetalhesViagem(
        @Path("storeId") storeId: String,
        @Path("viagemId") viagemId: String
    ): Response<DetalhesViagemResponse>

    // ========== GERAR PROTOCOLO ==========
    @POST("web/recebimento/{storeId}")
    suspend fun gerarProtocolo(
        @Path("storeId") storeId: String,
        @Body request: ProtocoloRequest
    ): Response<ProtocoloResponse>

    // ========== IMPRIMIR VIAGEM ==========
    @GET("web/recebimento/print/{storeId}/{viagemId}")
    suspend fun imprimirViagem(
        @Path("storeId") storeId: String,
        @Path("viagemId") viagemId: String
    ): Response<ImpressaoResponse>

    // ========== GERAR EXCEL DA VIAGEM ==========
    @GET("web/recebimento/planilha/{storeId}/{viagemId}")
    @Streaming
    suspend fun gerarExcelViagem(
        @Path("storeId") storeId: String,
        @Path("viagemId") viagemId: String
    ): Response<ResponseBody>
}

File: flutter/lib/network/api_service.dart
import 'dart:typed_data';
import '../../models/recebimento.dart';
import 'api_client.dart';

class ApiService {
  final ApiClient client;

  ApiService(this.client);

  Future<RecebimentoResponse> getRecebimentos({
    required String storeId,
    required String status,
    String? search,
    String sort = "asc",
    int page = 1,
  }) async {
    final query = <String, String>{
      'status': status,
      'sort': sort,
      'page': page.toString(),
    };
    if (search != null && search.isNotEmpty) query['search'] = search;
    final json = await client.get('web/recebimento/$storeId',
        query: query) as Map<String, dynamic>;
    return RecebimentoResponse.fromJson(json);
  }

  Future<DetalhesViagemResponse> getDetalhesViagem(
      String storeId, String viagemId) async {
    final json = await client.get('web/recebimento/$storeId/$viagemId')
        as Map<String, dynamic>;
    return DetalhesViagemResponse.fromJson(json);
  }

  Future<ProtocoloResponse> gerarProtocolo(
      String storeId, ProtocoloRequest request) async {
    final json = await client.post('web/recebimento/$storeId',
        body: request.toJson()) as Map<String, dynamic>;
    return ProtocoloResponse.fromJson(json);
  }

  Future<ImpressaoResponse> imprimirViagem(
      String storeId, String viagemId) async {
    final json = await client.get('web/recebimento/print/$storeId/$viagemId')
        as Map<String, dynamic>;
    return ImpressaoResponse.fromJson(json);
  }

  Future<Uint8List> gerarExcelViagem(
      String storeId, String viagemId) async {
    final bytes = await client.getBytes(
        'web/recebimento/planilha/$storeId/$viagemId');
    return Uint8List.fromList(bytes);
  }
}
